### PR TITLE
Using the dlrn status report to detect failed builds.

### PR DESCRIPTION
**Pull request by fuzzball81**
To get failure information, parsing the status_report.csv file is
the easiest way to get a list of builds that are failing. The csv file
provides all of the data needed to generate information to report the
error.

The unit tests use both good and bad csv files generated from a dlrn
instance.

Signed-off-by: Jason Joyce <fuzzball81@gmail.com>

### CODE REVIEW

File: atkinson/dlrn/http_data.py [hunk 1]
```diff
@@ -124,25 +129,46 @@ def _fetch_commit(self):
...
+        Generator that parsers a CVS file and returns each row
+
+        :param str url: The url to fetch the CSV file to parse
+        :return: A row from the CSV file
+        """
+        data = _raw_fetch(url, self._logger)
+        if data is not None:
+            data = data.replace(' ', '_')
+            split_data = data.split()
+            reader = csv.DictReader(split_data)
```
Comment (jguiditta) [inline atkinson/dlrn/http_data.py:166]: Is this one where it would make sense to use a 'with' statement, like open?

**Review by jguiditta — COMMENTED** [review]
LGTM, one question inline, otherwise, I'll approve

File: atkinson/dlrn/http_data.py (hunk 1 above)
Comment (fuzzball81) [reply atkinson/dlrn/http_data.py:166]: I wish the csv,DictReader could be used with like open, unfortunately it does not appear to have the needed __enter__ and __exit__ methods. See https://stackoverflow.com/questions/441130/using-with-statement-for-csv-files-in-python

Comment (jguiditta) [reply atkinson/dlrn/http_data.py:166]: Fair enough.

**Review by jguiditta — APPROVED** [review]

**Pull request merged**